File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    c = Counter(tangerine)
    k = len(tangerine) - k
    
    for num in sorted(c.keys(), key=lambda x:c[x]):
        if c[num] > k or k == 0:
            break
        k -= c[num]
        c.pop(num)
    

    return len(c)

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> counter = new HashMap<>();
        
        for (int tang : tangerine) {
            counter.merge(tang, 1, (o,v) -> o+1);
        }
        
        int answer = counter.keySet().size();
        
        List<Integer> c = counter.entrySet()
                                .stream()
                                .map(kv -> kv.getValue())
                                .sorted(Comparator.reverseOrder())
                                .collect(Collectors.toList());
        
        for (int i = 0; i < c.size(); i++) {        
            if (k <= 0) {
                return i;
            }
            
            k -= c.get(i);
        }
        
        
        return c.size();
    }
}